p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/transport/FtpTransportProtocolHandler.java

```diff
@@ -160,9 +159,8 @@ private FTPClient getClient(final URI uri) throws IOException {
         }
 
         if (!FTPReply.isPositiveCompletion(client.getReplyCode())) {
```

Comment:
Maybe one can have method (that possibly throws IOException) checkResult(client, message_with_placeholder)